### PR TITLE
RKE1 Agent configuration

### PR DESCRIPTION
**Pull request by mantis-toboggan-md**
https://github.com/rancher/dashboard/issues/8534

Updated to include https://github.com/rancher/dashboard/issues/8535 - agent configuration for EKS hosted and imported clusters. 

This should more or less line up with the RKE2 implementation https://github.com/rancher/dashboard/pull/8643

For more details on desired behavior check out [RFC](https://confluence.suse.com/pages/viewpage.action?pageId=1185186511) and [Designs](https://confluence.suse.com/pages/viewpage.action?spaceKey=CU&title=Ability+to+configure+agent+deployments+per+cluster)

Screen Shot 2023-04-20 at 2 13 30 PM
Screen Shot 2023-04-20 at 2 13 58 PM
Screen Shot 2023-04-20 at 2 14 05 PM
Screen Shot 2023-04-20 at 2 14 18 PM
Screen Shot 2023-04-20 at 2 14 44 PM
Screen Shot 2023-04-20 at 2 34 41 PM

### CODE REVIEW

**Comment by mantis-toboggan-md** [comment]
Ahh I forgot to say in my PR that we're still waiting for backend support for this issue; you just to verify that the request is shaped properly. Sorry about that @codyrancher real oversight on my part.

I found a bug while demoing this work this morning so I'm going to convert the PR back to draft until I get that resolved

**Comment by mantis-toboggan-md** [comment]
Updated to fix mistake in nodeAffinity spec and tweak pod affinity namespace selector text

**Comment by mantis-toboggan-md** [comment]
@codyrancher the backend work for this has been merged: this PR should be good to review now. Since it was a relatively small change I've updated the PR to add the agent configuration options to EKS provisioning.

**Comment by rak-phillip** [comment]
I'm seeing some inconsistent styling in `Operator` and `Effect` when adding tolerations.

image

vs. Advanced Options

image

Tested on Firefox 113.0b9 (64-bit)

**Comment by mantis-toboggan-md** [comment]
I've pushed a fix for the padding inconsistency in tolerations:
Screen Shot 2023-05-09 at 1 23 59 PM

File: lib/shared/addon/components/form-node-affinity-k8s/component.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    return out
+  },
+
+  removeWeight: (term) => {
+    const out = {
+      _id: term._id,
+      ...term.preference
+    }
+
+    return out
```
Comment (rak-phillip) [inline lib/shared/addon/components/form-node-affinity-k8s/component.js:122]: Similarly, it looks like `out` isn't required
```suggestion
    return {
      _id: term._id,
      ...term.preference
    }
```

File: lib/shared/addon/components/form-node-affinity-k8s/component.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  layout,
+
+  TERM_PRIORITY,
+  nodeAffinity:     null,
+  editing:          false,
+  allTerms:         [],
+
+  /**
+ * this component renders one list for required & preferred arrays of node selector terms
+ * each nodeaffinitytermk8s component can change between required and perferred
```
Comment (rak-phillip) [inline lib/shared/addon/components/form-node-affinity-k8s/component.js:20]:
```suggestion
 * each nodeaffinitytermk8s component can change between required and preferred
```

File: lib/shared/addon/components/form-pod-affinity-k8s/component.js [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+  layout,
+
+  TERM_PRIORITY,
+  podAffinity:     null,
+  podAntiAffinity: null,
+  editing:         false,
+  allTerms:        [],
+  /**
+ * this component renders one list for required & preferred arrays of terms in affinity & antiAffinity
+ * each podaffinitytermk8s component can change between affinity and antiaffinity and between required and perferred
```
Comment (rak-phillip) [inline lib/shared/addon/components/form-pod-affinity-k8s/component.js:24]:
```suggestion
 * each podaffinitytermk8s component can change between affinity and antiaffinity and between required and preferred
```

File: lib/shared/addon/components/form-pod-affinity-k8s/component.js [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+    return out
+  },
+
+  removeWeight: (term) => {
+    const out = {
+      _id: term._id,
+      ...term.podAffinityTerm
+    }
+
+    return out
```
Comment (rak-phillip) [inline lib/shared/addon/components/form-pod-affinity-k8s/component.js:202]: It doesn't look like `out` is required

```suggestion
    return {
      _id: term._id,
      ...term.podAffinityTerm
    }
```

**Review by rak-phillip — COMMENTED** [review]
Looks good overall. I ran through some tests to assert that basic functionality is in place. Nancy and I met out of band to step through and review some of the code within in this PR.

Any feedback I left is fairly minor. Feel free to address as you see fit.

**Review by rak-phillip — APPROVED** [review]
LGTM

**Pull request merged**